File: BiggerPython/GUITests.py
import pdbmolecules
import oclconfiguration
import geomutils
import molutils
import bogie
import linegrids
import basetypes
import dockdomains

# Tests about loading file and Bigger execution


def LoadTest():
    mol = FMolecules.LoadLayer('../PDB/4a0q.pdb')
    print('FName: ' + mol.FName)
    print('FID: ' + str(mol.FID))
    if mol.FParent is not None:
        print('FParent: ' + mol.FParent.FName)
    else:
        print('FParent: None')

    print('')
    print('Atoms:')
    print(len(mol.FAtoms))
    if len(mol.FAtoms) > 0:
        atom = mol.FAtoms[0]
        print('First Atom:')
        print('AtomName = ' + atom.AtomName)
        print('FID = ' + str(atom.FID))
        if atom.FParent is not None:
            print('FParent = ' + atom.FParent.FName)
        else:
            print('FParent = None')
        print('FAtomicNumber = ' + str(atom.FAtomicNumber))
        print('FCoord = ' + str(atom.FCoord[0]) + ' ' + str(atom.FCoord[1]) + ' ' + str(atom.FCoord[2]))
        print('FRadius = ' + str(atom.FRadius))
        print('FCharge = ' + str(atom.FCharge))
        print('FMass = ' + str(atom.FMass))
        print('Tag = ' + str(atom.Tag))

    print('')
    print('FGroups:')
    print(len(mol.FGroups))
    for f in range(len(mol.FGroups)):
        print('   ' + mol.FGroups[f].FName)

    print('')
    print('FBondsTable')
    print(len(mol.FBondsTable))
    if len(mol.FBondsTable) > 0:
        bond = mol.FBondsTable[0]
        print('First Bond:')
        print('Atom1: ' + bond.Atom1.FName)
        print('Atom2: ' + bond.Atom2.FName)
        print('BondType: ' + str(bond.BondType))
        print('Tag: ' + str(bond.Tag))

    print('')
    print('FType: ' + mol.FType)


def BiggerTest():
    target = FMolecules.LoadLayer('../PDB/3f6u.pdb')
    probe = FMolecules.LoadLayer('../PDB/4a0q.pdb')
    target.Transform(geomutils.Simmetric(molutils.FindCenter(target)))
    probe.Transform(geomutils.Simmetric(molutils.FindCenter(probe)))
    targetrads = geomutils.Add(molutils.ListRadii(target), 1.4)
    targetcoords = molutils.ListCoords(target)
    proberads = geomutils.Add(molutils.ListRadii(probe), 1.4)
    probecoords = molutils.ListCoords(probe)

    models = bogie.TModelManager(100, 300, [])
    models.GridScale = 1
    targetgrid = linegrids.TDockingGrid(1)
    targetgrid.BuildFromSpheres(targetcoords, targetrads)

    domain = None
    tick1 = 0
    MaxIters = 1
    for f in range(1, MaxIters + 1):
        tick1 = basetypes.GetTickCount()
        probegrid = linegrids.TDockingGrid(1)
        probegrid.BuildFromSpheres(probecoords, proberads)

        domain = dockdomains.TDockDomain(targetgrid, probegrid, 0)
        domain.MinimumOverlap = models.FMinOverlap
        domain.AssignModelManager(models.AddModel)
        domain.RemoveCores = True
        domain.BuildInitialDomain()
        domain.Score()
        print(str(models.FModels[0].OverlapScore) + ' (' + str(models.FModels[0].TransVec[0]) + ',' +
              str(models.FModels[0].TransVec[1]) + ',' + str(models.FModels[0].TransVec[2]) + ')')
    tick2 = basetypes.GetTickCount()

    domain.CalcDomainStats()
    print((tick2 - tick1) / 1000)
    print(str(domain.FDomainGrid.Shape.TotalCount) + ' cells')
    print(str(len(targetcoords)) + ' atoms')
    for f in range(len(models.FModels)):
        print(str(models.FModels[f].OverlapScore) + ' (' + str(models.FModels[f].TransVec[0]) + ', ' +
              str(models.FModels[f].TransVec[1]) + ', ' + str(models.FModels[f].TransVec[2]) + ')')
    print('')


oclconfiguration.DefaultConfig()
oclconfiguration.LoadAtomData()
oclconfiguration.LoadAAData()
FMolecules = pdbmolecules.TPDBModelMan(oclconfiguration.Config.MonomersPath)

# LoadTest()
BiggerTest()

File: TestChemera/guitests.pas
unit guitests;

{$mode objfpc}{$H+}

interface

uses
  Classes, SysUtils, oclconfiguration, chemeramain, pdbmolecules, molecules,
  basetypes, geomutils, bogie, linegrids, dockdomains, molutils, LCLProc;

procedure StartGUITest();
procedure LoadTest();
procedure BiggerTest();

var
  FMolecules:TPdbModelMan;

implementation

procedure StartGUITest();
begin
  LoadAtomData();
  LoadAAData();
  FMolecules:=TPdbModelMan.Create(Config.MonomersPath);

  // LoadTest();
  BiggerTest();
end;

procedure LoadTest();
var mol:TMolecule; atom:TAtom; bond:TAtomBond; f:integer; startTick,endTick:DWORD;

begin
  startTick:=GetTickCount;
  mol:=FMolecules.LoadLayer('../PDB/4a0q.pdb');
  WriteLn('FName: ' + mol.Name);
  WriteLn('FID: ' + IntToStr(mol.ID));
  if mol.Parent <> Nil then
    WriteLn('FParent: ' + mol.Parent.Name)
  else
    WriteLn('FParent: None');

  WriteLn('');
  WriteLn('Atoms:');
  WriteLn(Length(mol.GroupAtoms));
  if Length(mol.GroupAtoms) > 0 then
  begin
    atom:=mol.GroupAtoms[0];
    WriteLn('First Atom:');
    WriteLn('AtomName = ' + atom.Name);
    WriteLn('FID = ' + IntToStr(atom.ID));
    if atom.Parent <> Nil then
      WriteLn('FParent = ' + atom.Parent.Name)
    else
      WriteLn('FParent = None');
    WriteLn('FAtomicNumber = ' + IntToStr(atom.AtomicNumber));
    WriteLn('FCoord = ' + FloatToStr(atom.Coords[0]) + ' ' + FloatToStr(atom.Coords[1]) + ' ' + FloatToStr(atom.Coords[2]));
    WriteLn('FRadius = ' + FloatToStr(atom.Radius));
    WriteLn('FCharge = ' + FloatToStr(atom.Charge));
    WriteLn('FMass = ');
    WriteLn('Tag = ' + IntToStr(atom.Tag));
  end;

  WriteLn('');
  WriteLn('FGroups:');
  WriteLn(Length(mol.Groups));
  for f:=0 to High(mol.Groups) do
  begin
    WriteLn('   ' + mol.Groups[f].Name);
  end;

  WriteLn('');
  WriteLn('FType: ' + mol.MolType);

  endTick:=GetTickCount;
  WriteLn('Execution time: ' + FloatToStr((endTick - startTick) / 1000) + ' seconds');
end;

procedure BiggerTest();
var targetrads,proberads:TFloats;
  targetcoords,probecoords:TCoords;
  targetgrid,probegrid:TDockingGrid;
  domain:TDockDomain;
  tick1,tick2,startTick,endTick:DWORD;
  models:TModelManager;
  f:Integer;
  target,probe:TMolecule;
  MaxIters:Integer;

begin
  target:=FMolecules.LoadLayer('..\PDB\3f6u.pdb');
  probe:=FMolecules.LoadLayer('..\PDB\4a0q.pdb');
  startTick:=GetTickCount;

  target.Transform(Simmetric(FindCenter(target)));
  probe.Transform(Simmetric(FindCenter(probe)));

  targetrads:=Add(ListRadii(target),1.4);
  targetcoords:=ListCoords(target);

  proberads:=Add(ListRadii(probe),1.4);
  probecoords:=ListCoords(probe);

  models:=TModelManager.Create(100,300, nil);
  models.GridScale:=1;
  Writeln('Workin...');
  targetgrid:=TDockingGrid.Create(1);
  targetgrid.BuildFromSpheres(targetcoords,targetrads);

  MaxIters:=1;

  for f:=1 to MaxIters do
    begin
    tick1:=GetTickCount;
    probegrid:=TDockingGrid.Create(1);
    probegrid.BuildFromSpheres(probecoords,proberads);

    domain:=TDockDomain.Create(targetgrid,probegrid,0);
    domain.MinimumOverlap:=models.MinimumOverlap;
    domain.AssignModelManager(@models.AddModel);
    domain.RemoveCores:=True;
    domain.BuildInitialDomain;
    domain.Score;
      writeLn(models.Models[0].OverlapScore,' (',models.Models[0].TransVec[0],',',
              models.Models[0].TransVec[1],',',models.Models[0].TransVec[2],')');
    if f<MaxIters then
      begin
      domain.free;
      probegrid.free;
      end;
    end;
  tick2:=GetTickCount;

  domain.CalcDomainStats;

  WriteLn(FloatToStrF((tick2-tick1)/1000,ffFixed,4,3));
  WriteLn(domain.Size,' cells');
  DebugLn(IntToStr(Length(targetcoords)),' atoms');
  for f:=0 to 20 do
    with models.Models[f] do
      writeLn(OverlapScore,' (',TransVec[0],',',TransVec[1],',',TransVec[2],')');

  endTick:=GetTickCount;
  WriteLn('Execution time: ' + FloatToStr((endTick - startTick) / 1000) + ' seconds');
end;

end.

